### PR TITLE
Dm 3954: Add a PageBuilder Publications Component and Simple Button Component

### PR DESCRIPTION
**Pull request by camillevilla**
### JIRA issue link
[DM-3954](https://agile6.atlassian.net/browse/DM-3954)

## Description - what does this code do?
- Adds a publication component
  - fields: title, url OR attachment, published_in, publication date, title
  - paginates for 10+ components
  - applies card styling for <2 publications 
    - [x] TODO: check with Becca re: intended behavior for 1 publication
- Adds a simple button component to support VA immersive homepage design (i.e. See more button that links to Publications page - https://www.figma.com/file/ccwLbt94U9QGfnuFCwOXzY/Core-Designs---VA-DM?type=design&node-id=6181%3A38719&mode=design&t=GsnXX0gGDh83hjz2-1)

## Testing done - how did you test it/steps on how can another person can test it 
1. Log in as an admin, create a page group and test page
2. Create 13 with publications only titles. Save page.
3. Insert a Simple Button component between publications 2 and 3. Save page.
4. View page. The first 2 components should have card styling applied. The latter publications should have single-column list styling and paginate after 10 publications.
5. Click the "Load more" button to confirm pagination is working
6. Add a URL to one of the publications. Confirm that the title is now a link.
7. Add an attachment to one of the publications. Confirm that the title opens the uploaded file.

## Screenshots, Gifs, Videos from application (if applicable)
Screenshot 2023-07-11 at 5 46 19 PM

## Link to mock-ups/mock ups (image file if you have it) (if applicable)
- Featured Publications (2 publications on homepage): https://www.figma.com/file/ccwLbt94U9QGfnuFCwOXzY/Core-Designs---VA-DM?type=design&node-id=6181%3A38719&mode=design&t=GsnXX0gGDh83hjz2-1
- Publications page: https://www.figma.com/file/ccwLbt94U9QGfnuFCwOXzY/Core-Designs---VA-DM?type=design&node-id=6181%3A38985&mode=design&t=GsnXX0gGDh83hjz2-1 

## Acceptance criteria
- [ ]

## Definition of done
- [x] Unit tests written (if applicable)
- [x] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating JIRA issue
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/assets/javascripts/_page_show.es6
Comment (camillevilla) [inline app/assets/javascripts/_page_show.es6:1]: A little more context on signed resources: https://github.com/department-of-veterans-affairs/diffusion-marketplace/pull/554

File: app/assets/javascripts/active_admin.js
Comment (camillevilla) [inline app/assets/javascripts/active_admin.js:1]: Er, whoops. Looks like my editor did a bunch of of whitespace formatting on this file.

File: app/views/page/show.html.erb [hunk 1]
```diff
@@ -232,6 +232,44 @@
...
+            <% if @publication_ids.include?(component.id) %>
+              <%
+                component_index = @publication_list_component_index
+                plc = @publication_list_components[component_index]
+                @publication_list_component_index = component_index + 1
+              %>
+              <% if plc[:pagy].count <= 2 # use card styling for 2 or fewer  %>
+                <ul class="publication-component-list-cards usa-card-group dm-paginated-<%= component_index %>-events margin-bottom-4 <%= page_narrow_classes %><%= ' margin-bottom-0' if last_component === index %>">
+                  <%= render(partial: 'publications_list_cards', locals: {publications: plc[:publications]}) %>
+                </ul>
```
Comment (camillevilla) [inline app/views/page/show.html.erb:247]: 2-card styling for VA Immersive homepage: https://www.figma.com/file/ccwLbt94U9QGfnuFCwOXzY/Core-Designs---VA-DM?type=design&node-id=6181%3A38719&mode=design&t=GsnXX0gGDh83hjz2-1

**Review by PhilipDeFraties — APPROVED** [review]
QA steps worked as described and code lgtm. Just wondering, should I be locally testing a publications page (VA Immersive?) in addition to testing the home page functionality?

**Comment by camillevilla** [comment]
> QA steps worked as described and code lgtm. Just wondering, should I be locally testing a publications page (VA Immersive?) in addition to testing the home page functionality?

Ah, no need. Testing on the same page will suffice for these. And we get the added bonus of making sure there can be two different publication lists on the same page

**Pull request merged**